.github/workflows/publish_dockerhub.yml

```diff
@@ -0,0 +1,50 @@
+
+name: "Docker hub publish"
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+      generic_tag:
+        required: true
+        type: string
+
+jobs:
+  deploy-dockerhub:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download docker image
+        uses: actions/download-artifact@v4
+        with:
+          name: image
+          path: /tmp
+
+      # setup containerd to preserve provenance attestations :https://docs.docker.com/build/attestations/#creating-attestations
+      - name: Setup docker with containerd
+        uses: crazy-max/ghaction-setup-docker@v3
+        with:
+          daemon-config: |
+            {
+              "features": {
+                "containerd-snapshotter": true
```

Comment:
why is this `containerd-snapshotter` thing required?

Reply:
Without it, SBOM and provenance attestation are not preserved.

I just gave it a test. 

With containerd activated : 

```
docker buildx imagetools inspect ghcr.io/giom-l/kafka-ui:main --format '{{ json .Provenance }}'
{
  "linux/amd64": {
    "SLSA": {
      "buildType": "https://mobyproject.org/buildkit@v1",
      "builder": {
        "id": "https://github.com/giom-l/kafka-ui/actions/runs/9396399214"
      },
      "invocation": {
        "configSource": {
          "entryPoint": "Dockerfile"
        },
        "environment": {
          "platform": "linux/amd64"
        },
        "parameters": {
          "frontend": "dockerfile.v0",
          "locals": [
            {
              "name": "context"
            },
            {
              "name": "dockerfile"
            }
          ]
        }
      },
      "materials": [
        {
          "digest": {
            "sha256": "5ba01ac30fcc3d7be766ad55dae06de90b15fe0d940baac94b23ddb231e8b089"
          },
          "uri": "pkg:docker/azul/zulu-openjdk-alpine@17.0.11-jre-headless?platform=linux%2Famd64"
        },
        {
          "digest": {
            "sha256": "176e0869c38aeaede37e594fcf182c91d44391a932e1d71e99ec204873445a33"
          },
          "uri": "pkg:docker/docker/buildkit-syft-scanner@stable-1"
        }
      ],
      "metadata": {
        "buildFinishedOn": "2024-06-06T06:54:26.509711264Z",
        "buildInvocationID": "pfnymph4d1p9ra92kr3ktm7dv",
        "buildStartedOn": "2024-06-06T06:54:08.308818665Z",
        "completeness": {
          "environment": true,
          "materials": false,
          "parameters": false
        },
        "https://mobyproject.org/buildkit@v1#metadata": {
          "vcs": {
            "localdir:context": "api",
            "localdir:dockerfile": "api",
            "revision": "48d353268657ebb2cc61004f2b59fbe157be8383",
            "source": "https://github.com/giom-l/kafka-ui"
          }
        },
        "reproducible": false
      }
    }
  },
  "linux/arm64": {
    "SLSA": {
      "buildType": "https://mobyproject.org/buildkit@v1",
      "builder": {
        "id": "https://github.com/giom-l/kafka-ui/actions/runs/9396399214"
      },
      "invocation": {
        "configSource": {
          "entryPoint": "Dockerfile"
        },
        "environment": {
          "platform": "linux/amd64"
        },
        "parameters": {
          "frontend": "dockerfile.v0",
          "locals": [
            {
              "name": "context"
            },
            {
              "name": "dockerfile"
            }
          ]
        }
      },
      "materials": [
        {
          "digest": {
            "sha256": "5ba01ac30fcc3d7be766ad55dae06de90b15fe0d940baac94b23ddb231e8b089"
          },
          "uri": "pkg:docker/azul/zulu-openjdk-alpine@17.0.11-jre-headless?platform=linux%2Farm64"
        },
        {
          "digest": {
            "sha256": "176e0869c38aeaede37e594fcf182c91d44391a932e1d71e99ec204873445a33"
          },
          "uri": "pkg:docker/docker/buildkit-syft-scanner@stable-1"
        }
      ],
      "metadata": {
        "buildFinishedOn": "2024-06-06T06:54:26.509711264Z",
        "buildInvocationID": "pfnymph4d1p9ra92kr3ktm7dv",
        "buildStartedOn": "2024-06-06T06:54:08.308818665Z",
        "completeness": {
          "environment": true,
          "materials": false,
          "parameters": false
        },
        "https://mobyproject.org/buildkit@v1#metadata": {
          "vcs": {
            "localdir:context": "api",
            "localdir:dockerfile": "api",
            "revision": "48d353268657ebb2cc61004f2b59fbe157be8383",
            "source": "https://github.com/giom-l/kafka-ui"
          }
        },
        "reproducible": false
      }
    }
  }
}%                                 
```

Exact same workflow, just without containerd

```
docker buildx imagetools inspect ghcr.io/giom-l/kafka-ui:main --format '{{ json .Provenance }}'
{}%                                                                                                                                                
```

Reply:
thanks!